server/modules/search/search.controller.ts

```
@@ -0,0 +1,259 @@
+
+/**
+ * Main search endpoint
+ * Handles comprehensive search with pagination
+ */
+export const search = async (req: Request, res: Response) => {
+  try {
+    const { query, location, sport, page = "1", limit = "10" } = req.query;
+    const searchQuery = (query as string) || "";
+    const sportQuery = (sport as string) || "";
+    const locationQuery = (location as string) || "";
+    const pageNum = parseInt(page as string) || 1;
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)